lib/puppet/util/selinux.rb

```diff
@@ -65,13 +65,18 @@ def parse_selinux_context(component, context)
     unless context =~ /^([^\s:]+):([^\s:]+):([^\s:]+)(?::([\sa-zA-Z0-9:,._-]+))?$/
       raise Puppet::Error, "Invalid context to parse: #{context}"
     end
-    ret = {
-      :seluser => $1,
-      :selrole => $2,
-      :seltype => $3,
-      :selrange => $4,
-    }
-    ret[component]
+    case component
+    when :seluser
+      $1
```

Comment:
My main goal with this quick refactor was to get rid of the extra hash allocation/lookup while I was in here, but I no longer actually need to change this method in this commit. I'll pull this out and push up a separate maint commit (still part of this PR, probably) that will clean up this function a little more than just the hash->case change